### PR TITLE
Fix wire handling on special control wires

### PR DESCRIPTION
**Pull request by albi3ro**
**Context:**

A user was doing `qml.CRY(0.1, wires=((0,1), 2)`.  In this case, the control wire should be a single wire with a value `(0,1)`.  Yes, this is weird, but this is how wires work in pennylane.  Instead, it was being interpretted as having two control wires, even though `CRY` is only for singly-controlled `RY` gates.

**Description of the Change:**

Some minor slicing changes in how we handle special controlled ops.  

Now we have:
```
>>> qml.CZ(wires=((0,1), 2))
CZ(wires=[(0, 1), 2])
```

Which is consistent with how we treat wires elsewhere in pennylane.

**Benefits:**

Don't end up with a two qubit gate on three qubits.

**Possible Drawbacks:**

Still weird, and still not what the user was looking for when they provided wires like that.  But at least now they'll get a better error.

**Related GitHub Issues:**

### CODE REVIEW

**Review by astralcai — APPROVED** [review]

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -341,6 +341,8 @@
 
 <h3>Bug fixes 🐛</h3>
 
+* Fixes a bug in the wire handling on special controlled ops.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:344]:
```suggestion
* Fixes a bug in the wire handling on special controlled ops.
  [(#5856)](https://github.com/PennyLaneAI/pennylane/pull/5856)
```

File: pennylane/ops/op_math/controlled_ops.py [hunk 2]
```diff
@@ -791,8 +789,7 @@ def _primitive_bind_call(cls, wires, id=None):
         return cls._primitive.bind(*wires, n_wires=2)
 
     def __init__(self, wires, id=None):
-        control_wire, wire = wires
-        super().__init__(qml.PauliX(wires=wire), control_wire, id=id)
+        super().__init__(qml.PauliX(wires=wires[1:]), wires[:1], id=id)
```
Comment (dwierichs) [inline pennylane/ops/op_math/controlled_ops.py:792]: Could make this consistent with Y, Z
```suggestion
        super().__init__(qml.X(wires=wires[1:]), wires[:1], id=id)
```

**Review by dwierichs — APPROVED** [review]
Thanks for the quick fix! :rocket:

File: pennylane/ops/op_math/controlled_ops.py (hunk 2 above)
Comment (albi3ro) [reply pennylane/ops/op_math/controlled_ops.py:792]: don't really want to restart all the CI...

**Pull request merged**